src/openms_gui/source/VISUAL/DIALOGS/TheoreticalSpectrumGenerationDialog.cpp

```diff
@@ -95,16 +113,16 @@ namespace OpenMS
     // select peptide sequence by default and update interface
     ui_->seq_type->setCurrentText("Peptide");
     seqTypeSwitch_();
```

Comment:
will this be called twice (because you call `setCurrentText()`) which is connected to seqTypeSwitch_?!

Reply:
That's what I also would've expected but I think the signal-slot connection only work after object construction. Therefore `seqTypeSwitch_` needs to be called here.

Reply:
The slot `currentTextChanged` only gets called if the text actually changes. That's why calling the slot here is necessary. However I removed setting the current text since it is not really needed. (The ui-default for this combo box is already 'Peptide')